src/mapblock.cpp

```diff
@@ -490,7 +462,12 @@ void MapBlock::deSerialize(std::istream &in_compressed, u8 version, bool disk)
 
 	u8 flags = readU8(is);
 	is_underground = (flags & 0x01) != 0;
-	m_day_night_differs = (flags & 0x02) != 0;
+	if (version >= 30) {
+		m_is_air = (flags & 0x02) != 0;
+	} else {
+		m_is_air = false;
+		m_is_air_expired = true;
```

Comment:
for blocks from disk this is actually very easy: check if the node-ID mapping contains only one entry and it's `"air"`